h2o-genmodel/src/main/java/hex/genmodel/tools/PrintMojo.java

```diff
@@ -0,0 +1,168 @@
+    System.out.println("");
+    System.out.println("Example:");
+    System.out.println("");
+    System.out.println("    (brew install graphviz)");
+    System.out.println("    java -cp h2o.jar hex.genmodel.tools.PrintMojo --tree 0 -i model_mojo.zip -o model.gv");
+    System.out.println("    dot -Tpng model.gv -o model.png");
+    System.out.println("    open model.png");
+    System.out.println("");
+    System.exit(1);
+  }
+
+  private void parseArgs(String[] args) {
+    try {
+      for (int i = 0; i < args.length; i++) {
+        String s = args[i];
+        switch (s) {
+          case "--tree":
+            i++;
+            if (i >= args.length) usage();
+            s = args[i];
+            try {
+              treeToPrint = Integer.parseInt(s);
+            }
+            catch (Exception e) {
+              System.out.println("ERROR: invalid --tree argument (" + s + ")");
+              System.exit(1);
+            }
+            break;
+
+          case "--levels":
+            i++;
+            if (i >= args.length) usage();
+            s = args[i];
+            try {
+              maxLevelsToPrintPerEdge = Integer.parseInt(s);
+            }
+            catch (Exception e) {
+              System.out.println("ERROR: invalid --levels argument (" + s + ")");
+              System.exit(1);
+            }
+            break;
+
+          case "--input":
+          case "-i":
+            i++;
+            if (i >= args.length) usage();
+            s = args[i];
+            loadMojo(s);
+            break;
+
+          case "--raw":
+            printRaw = true;
+            break;
+
+          case "-o":
+          case "--output":
+            i++;
+            if (i >= args.length) usage();
+            outputFileName = args[i];
+            break;
+
+          default:
+            System.out.println("ERROR: Unknown command line argument: " + s);
+            usage();
+            break;
+        }
+      }
+    } catch (Exception e) {
+      e.printStackTrace();
+      usage();
+    }
+  }
+
+  private void validateArgs() {
+    if (genModel == null) {
+      System.out.println("ERROR: Must specify -i");
+      usage();
+    }
+  }
+
+  private void run() throws Exception {
+    validateArgs();
+
+    PrintStream os;
+    if (outputFileName != null) {
+      os = new PrintStream(new FileOutputStream(new File(outputFileName)));
+    }
+    else {
+      os = System.out;
+    }
+
+    if (genModel instanceof GbmMojoModel) {
+      Graph g = ((GbmMojoModel) genModel).computeGraph(treeToPrint);
+      if (printRaw) {
+        g.print();
+      }
+      g.printDot(os, maxLevelsToPrintPerEdge);
+    }
+    else if (genModel instanceof DrfMojoModel) {
+      Graph g = ((DrfMojoModel) genModel).computeGraph(treeToPrint);
+      if (printRaw) {
+        g.print();
+      }
+      g.printDot(os, maxLevelsToPrintPerEdge);
+    }
+    else {
+      System.out.println("ERROR: Unknown MOJO type");
+      System.exit(1);
+    }
```

Comment:
If you move the `.computeGraph(treeToPrint)` call into the base `SharedTreeMojoModel` (using the `._ntrees_per_class` variable), then these if branches will collapse into a single one.